File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    int a, b, c;

    while (true) {
        cin >> a >> b >> c;
        if (a == 0 && b == 0 && c == 0) { break; }

        int maxSide = max(a, max(b, c));
        int sumOther = a + b + c - maxSide;

        if (maxSide >= sumOther) {
            cout << "Invalid" << endl;
        } else if (a == b && b == c && c == a) {
            cout << "Equilateral" << endl;
        } else if (a == b || b == c || c == a) {
            cout << "Isosceles" << endl;
        } else {
            cout << "Scalene" << endl;
        }
    }

    return 0;
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys

def classify(a: int, b: int, c: int) -> str:
    sides = sorted((a, b, c))
    if sides[2] >= sides[0] + sides[1]:
        return 'Invalid'
    if sides[0] == sides[1] == sides[2]:
        return 'Equilateral'
    if sides[0] == sides[1] or sides[1] == sides[2] or sides[0] == sides[2]:
        return 'Isosceles'
    return 'Scalene'

def main() -> None:
    results = []
    for line in sys.stdin:
        a, b, c = map(int, line.split())
        if a == b == c == 0:
            break
        results.append(classify(a, b, c))
    sys.stdout.write('
'.join(results))

if __name__ == '__main__':
    main()
